public_html/wp-content/themes/pattern-directory/src/components/pattern-thumbnail/canvas.js

```diff
@@ -40,32 +39,27 @@ export default function ( { url } ) {
 		return () => {
 			window.removeEventListener( 'resize', handleOnResize );
 		};
-	}, [] );
+	}, [ isVisible ] );
 
 	const style = {
 		border: 'none',
-		width: `${ VIEWPORT_WIDTH }px`,
+		width: '100%',
 		maxWidth: 'none',
-		height: `${ getCardFrameHeight( VIEWPORT_WIDTH ) }px`,
-		transform: `scale(${ frameScale })`,
-		transformOrigin: isRTL() ? 'top right' : 'top left',
-		pointerEvents: 'none',
+		height: `${ frameHeight }px`,
+		display: 'flex',
+		alignItems: 'center',
+		justifyContent: 'center',
 	};
 
 	return (
-		<div
-			ref={ wrapperRef }
-			style={ {
-				height: frameHeight,
-				overflow: 'hidden',
-			} }
-		>
-			<iframe
+		<div ref={ wrapperRef }>
+			<Screenshot
 				className="pattern-grid__preview"
-				title={ __( 'Pattern Preview', 'wporg-patterns' ) }
-				tabIndex="-1"
+				alt={ alt || __( 'Pattern Preview', 'wporg-patterns' ) }
 				style={ style }
-				src={ shouldLoad ? url : '' }
+				size={ { width: VIEWPORT_WIDTH, height: getCardFrameHeight( VIEWPORT_WIDTH ) } }
+				isReady={ shouldLoad }
+				src={ url }
```

Comment:
Can patterns be updated? 

If so you'll want to introduce a way to break the cache so a new mShot is generated.

Reply:
Yeah, they can, good catch — I added the pattern's modified date as part of the URL.